janus_core/training/preprocess.py

```diff
@@ -0,0 +1,88 @@
+"""Preprocess MLIP training data."""
+
+from collections.abc import Sequence
+from typing import Any, Optional
+
+from mace.cli.preprocess_data import run
+from mace.tools import build_preprocess_arg_parser as mace_parser
+import yaml
+
+from janus_core.helpers.janus_types import PathLike
+from janus_core.helpers.log import config_logger, config_tracker
+from janus_core.helpers.utils import check_files_exist, none_to_dict
+
+
+def preprocess(
+    mlip_config: PathLike,
+    req_file_keys: Optional[Sequence[PathLike]] = None,
+    attach_logger: bool = False,
+    log_kwargs: Optional[dict[str, Any]] = None,
+    track_carbon: bool = True,
+    tracker_kwargs: Optional[dict[str, Any]] = None,
+) -> None:
+    """
+    Convert training data to hdf5 by passing a configuration file to the MLIP's CLI.
+
+    Currently only supports MACE models, but this can be extended by replacing the
+    argument parsing.
+
+    Parameters
+    ----------
+    mlip_config : PathLike
+        Configuration file to pass to MLIP.
+    req_file_keys : Optional[Sequence[PathLike]]
+        List of files that must exist if defined in the configuration file.
+        Default is ("train_file", "test_file", "valid_file").
+    attach_logger : bool
+        Whether to attach a logger. Default is False.
+    log_kwargs : Optional[dict[str, Any]]
+        Keyword arguments to pass to `config_logger`. Default is {}.
+    track_carbon : bool
+        Whether to track carbon emissions of calculation. Default is True.
+    tracker_kwargs : Optional[dict[str, Any]]
+        Keyword arguments to pass to `config_tracker`. Default is {}.
+    """
+    (log_kwargs, tracker_kwargs) = none_to_dict((log_kwargs, tracker_kwargs))
```

Comment:
```suggestion
    log_kwargs, tracker_kwargs = none_to_dict((log_kwargs, tracker_kwargs))
```
Don't strictly need these for unpacking